File: java-bookstore/src/main/java/com/bookstore/entity/Contract.java
package com.bookstore.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.ToString;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Date;

@Entity
@Table(name = "contract")
@Data
@ToString
public class Contract {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String companyName;
    private String contractNum;
    private String clientName;

    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @JsonFormat(pattern = "dd/MM/yyyy")         //to accept the json string format
    private Date contractAt;
    private BigDecimal contractAmount;
    private String projectName;
    private String description;
    private Date createAt;
    private Date updateAt;
}

File: angular-bookstore/src/app/common/contract.ts
// "id" : 101,
// "clientName" : "北京AA有限公司",
// "contractNum" : "ZBS2021-001-006",
// "contractAt" : "2012-01-01",
// "contractAmount" : 6896830.00,
// "projectName" : "AAAAAAAAAAAAAAAA邑上苑热计量改造项目",
// "description" : "世AA邑上苑热计量改造项目",
// "createAt" : "2012-01-01",
// "updateAt" : "2012-01-01"

export class Contract {
    id: number;

    companyName: string;
    clientName: string;
    contractNum: string;

    contractAt: Date;
    contractAmount: number;
    projectName: string;
    description: string;

    createAt: Date;
    updateAt: Date;
}
